source/material_model/damage_rheology.cc

```diff
@@ -0,0 +1,2027 @@
+                                :
+                                1.0;
+
+      // Currently this will never be called without adiabatic_conditions initialized, but just in case
+      const double adiabatic_pressure = this->get_adiabatic_conditions().is_initialized()
+                                        ?
+                                        this->get_adiabatic_conditions().pressure(position)
+                                        :
+                                        pressure;
+
+      // find out in which phase we are
+      const unsigned int ol_index = get_phase_index(position, temperature, adiabatic_pressure);
+
+      // TODO: we use the prefactors from Behn et al., 2009 as default values, but their laws use the strain rate
+      // and we use the second invariant --> check if the prefactors should be changed
+      double energy_term = exp((diffusion_activation_energy[ol_index] + diffusion_activation_volume[ol_index] * adiabatic_pressure)
+                               / (diffusion_creep_exponent[ol_index] * constants::gas_constant * temperature));
+      if (this->get_adiabatic_conditions().is_initialized())
+        {
+          const double adiabatic_energy_term
+            = exp((diffusion_activation_energy[ol_index] + diffusion_activation_volume[ol_index] * adiabatic_pressure)
+                  / (diffusion_creep_exponent[ol_index] * constants::gas_constant * this->get_adiabatic_conditions().temperature(position)));
+
+          const double temperature_dependence = energy_term / adiabatic_energy_term;
+          if (temperature_dependence > max_temperature_dependence_of_eta)
+            energy_term = adiabatic_energy_term * max_temperature_dependence_of_eta;
+          if (temperature_dependence < 1.0 / max_temperature_dependence_of_eta)
+            energy_term = adiabatic_energy_term / max_temperature_dependence_of_eta;
+        }
+
+      const double strain_rate_dependence = (1.0 - diffusion_creep_exponent[ol_index]) / diffusion_creep_exponent[ol_index];
+
+      return pow(diffusion_creep_prefactor[ol_index],-1.0/diffusion_creep_exponent[ol_index])
+             * std::pow(second_strain_rate_invariant,strain_rate_dependence)
+             * pow(grain_size, diffusion_creep_grain_size_exponent[ol_index]/diffusion_creep_exponent[ol_index])
+             * energy_term;
+    }
+
+
+
+    template <int dim>
+    double
+    DamageRheology<dim>::
+    dislocation_viscosity (const double      temperature,
+                           const double      pressure,
+                           const std::vector<double> &composition,
+                           const SymmetricTensor<2,dim> &strain_rate,
+                           const Point<dim> &position,
+                           const double viscosity_guess) const
+    {
+      const double diff_viscosity = diffusion_viscosity(temperature,pressure,composition,strain_rate,position) ;
+
+      // Start the iteration with the full strain rate
```

Comment:
At some point, do we want to do the Newton iteration here that's implemented in the diffusion dislocation model? Could be more effective... But then it might not be that important, given that we don't have to compute the grain size change for this.

Reply:
I would prefer we leave this for an update. Who knows what that would do to the stability of the iteration.